### PR TITLE
Remove integrated REPL

### PR DESCRIPTION
**Pull request by slavapestov**
The integrated REPL has been deprecated in favor of the LLDB REPL for a while, and the implementation had accumulated some amount of technical debt throughout the code base. Since most people don't use it and are not even aware that it exists, we can remove it.

Note that this does not impact usage of the LLDB REPL (`lldb --repl` or just `swift`), Swift expression evaluation in LLDB, or the ability to run Swift scripts in immediate mode (`swift foo.swift`).

Forum discussion: https://forums.swift.org/t/rfc-removing-the-integrated-repl/35441

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
Well, I guess this won't work, because there's no way to just run lldb tests.

File: stdlib/public/Darwin/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,6 @@ endif()
 set(all_overlays
   Accelerate
   AppKit
-  ARKit
```
Comment (dcci) [inline stdlib/public/Darwin/CMakeLists.txt:19]: How come you're removing ARKit?

**Comment by dcci** [comment]
Nevermind, I guess this is because it doesn't compile.

**Comment by slavapestov** [comment]
https://github.com/apple/llvm-project/pull/1188
@swift-ci Please smoke test

**Comment by johanatan** [comment]
👍🏻

**Pull request merged**